modules/ppcp-button/src/Endpoint/ApproveOrderEndpoint.php

```diff
@@ -173,7 +183,7 @@ public function handle_request(): bool {
 				wp_send_json_success( $order );
 			}
 
-			if ( ! $order->status()->is( OrderStatus::APPROVED ) ) {
+			if ( $this->order_helper->contains_physical_goods( $order ) && ! $order->status()->is( OrderStatus::APPROVED ) ) {
```

Comment:
Why we don't need to check this status for virtual products? And why it was happening only for the 2nd order?

Reply:
@AlexP11223 because when using a vaulted payment for an order that does not contains physical goods (a virtual product in WC for exmaple), it receives the order status as `CREATED` instead of `APPROVED` here. The same happens here: https://github.com/woocommerce/woocommerce-paypal-payments/pull/674/files#diff-e1a4bb00801efe53b8a9341d53440edf1208f856bb0c68df685afca73a5de272R174